spec/support/shared_entity_specs.rb

```diff
@@ -133,3 +133,74 @@ def verify_signature(pubkey, signature, signed_string)
     end
   end
 end
+
+shared_examples "a JSON Entity" do
+  it "#to_json output matches JSON schema" do
+    json = described_class.new(data).to_json
+    expect(json.to_json).to match_json_schema(:entity_schema)
+  end
+
+  let(:to_json_output) { described_class.new(data).to_json.to_json }
+
+  it "contains described_class property matching the entity class (underscored)" do
+    expect(to_json_output).to include_json(entity_type: described_class.entity_name)
+  end
+
+  it "contains JSON properties for each of the entity properties with the entity_data property" do
+    entity_data = entity_hash_from(data)
+    entity_data.delete(:parent)
+    nested_elements = entity_data.select {|_key, value| value.is_a?(Array) || value.is_a?(Hash) }
+    entity_data.reject! {|_key, value| value.is_a?(Array) || value.is_a?(Hash) }
+
+    expect(to_json_output).to include_json(entity_data: entity_data)
+    nested_elements.each {|key, value|
+      type = described_class.class_props[key]
+      if value.is_a?(Array)
+        data = value.map {|element|
+          {
+            entity_type: type.first.entity_name,
+            entity_data: element
+          }
+        }
+        expect(to_json_output).to include_json(entity_data: {key => data})
+      else
+        expect(to_json_output).to include_json(
+          entity_data: {
+            key => {
+              entity_type: type.entity_name,
+              entity_data: value
+            }
+          }
+        )
+      end
+    }
+  end
+
+  include_examples "JSON is parsable with #from_json" do
+    let(:entity_class) { described_class }
+  end
+end
+
+shared_examples "JSON is parsable with #from_json" do
+  it "is parsable with #from_json" do
+    expect {
+      entity_class.from_json(JSON.parse(json))
+    }.not_to raise_error
+  end
+end
+
+shared_examples ".from_json return object verification" do
+  it "from_json(entity_json).to_json should match entity.to_json" do
+    entity_json = entity.to_json.to_json
+    expect(entity_class.from_json(JSON.parse(entity_json)).to_json.to_json).to eq(entity_json)
+  end
+end
+
+# TODO: move to a shared_parser_specs.rb?
```

Comment:
I thought about this, and I think it is cleaner to move it to it's own file. And an extra file for that doesn't hurt us :)